frontend/src/app/modules/challenges/online-editor/online-editor-page/online-editor-page.component.ts

```diff
@@ -0,0 +1,155 @@
+import { BreakpointObserver } from '@angular/cdk/layout';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
+import { ChallengeService } from '@core/services/challenge.service';
+import { languageNameMap } from '@shared/mappings/language-map';
+import { IChallenge } from '@shared/models/challenge/challenge';
+import { IChallengeVersion } from '@shared/models/challenge-version/challenge-version';
+import { EditorOptions } from '@shared/models/options/editor-options';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+    selector: 'app-online-editor-page',
+    templateUrl: './online-editor-page.component.html',
+    styleUrls: ['./online-editor-page.component.sass'],
+})
+export class OnlineEditorPageComponent implements OnDestroy, OnInit {
+    private destroyed$: Subject<void> = new Subject<void>();
+
+    activeTab: string = 'Description';
+
+    splitDirection: 'horizontal' | 'vertical';
+
+    splitLeftMinSize: number = 20;
+
+    splitRightMinSize: number = 20;
+
+    challenge: IChallenge;
+
+    selectedLanguage: string;
+
+    selectedLanguageVersion: string;
+
+    languages: string[];
+
+    languageVersions: string[];
+
+    initialSolution?: string;
+
+    testCode?: string;
+
+    editorOptions: EditorOptions;
+
+    constructor(
+        private activatedRoute: ActivatedRoute,
+        private challengeService: ChallengeService,
+        private breakpointObserver: BreakpointObserver,
+        private router: Router,
+    ) {
+        breakpointObserver
+            .observe(['(max-width: 843px)'])
+            .pipe(takeUntil(this.destroyed$))
+            .subscribe((result) => {
+                if (result.matches) {
+                    this.splitDirection = 'vertical';
+                } else {
+                    this.splitDirection = 'horizontal';
+                }
+            });
+    }
+
+    ngOnInit() {
+        this.splitDirection = 'horizontal';
+        this.activatedRoute.paramMap.subscribe((params: ParamMap) => {
+            const challengeId = +params.get('id')!;
```

Comment:
why do we need + here?

Reply:
That's my advice, so challengeId will be number, not string